src/components/Search/Search.jsx

```diff
@@ -1,31 +1,44 @@
 import React, { useState } from 'react';
 import SearchIcon from '../Icons/SearchIcon';
-import { ENTER_KEY } from '../../constants';
-import { SearchWrapper, SearchInput } from './styles';
+import moviesAPI from '../../api/movies';
+import List from './List';
+import { LIST_PAGE, LIST_PER_PAGE } from '../../constants';
+import { SearchWrapper, SearchInput, IconContainer } from './styles';
 
 const Search = () => {
-  const [value, setvalue] = useState('');
+  const [value, setValue] = useState('');
+  const [data, setData] = useState([]);
 
-  const onEnterPress = (e) => {
-    if (e.charCode === ENTER_KEY) {
-      setvalue(value);
+  const onHandleChange = async (e) => {
+    const currentValue = e.target.value;
+    setValue(currentValue);
+    if (!currentValue.length) {
+      setData([]);
+    }
+    if (currentValue.trim().length) {
+      const { movies } = await moviesAPI.get({
+        title: e.target.value,
+        page: LIST_PAGE,
+        perPage: LIST_PER_PAGE,
+      });
+      setData(movies);
     }
-  };
-
-  const onHandleChange = (e) => {
-    setvalue(e.target.value);
   };
 
   return (
-    <SearchWrapper>
-      <SearchInput
-        placeholder="Search…"
-        onKeyPress={onEnterPress}
-        onChange={onHandleChange}
-        value={value}
-      />
-      <SearchIcon />
-    </SearchWrapper>
+    <div>
+      <SearchWrapper>
+        <SearchInput
+          placeholder="Search…"
+          onChange={onHandleChange}
+          value={value}
+        />
+        <IconContainer>
+          <SearchIcon />
+        </IconContainer>
+      </SearchWrapper>
+      {value.length ? <List movies={data} /> : <div />}
```

Comment:
Check if `length` > 0 or cast to bool `!!`. Why do you need to render empty `<div>` here? You can rewrite this expression using binary operator, e.g.:
```js
{!!value.length && <List movies={data} />}
```